dissect/target/plugin.py

```diff
@@ -793,8 +796,123 @@ def _modulepath(cls) -> str:
+        """
+        super().__init__(target)
+
+        # The code below only applies to the direct subclass
+        # indirect subclasses are finished here.
+        if self.__class__ != self.__nsplugin__:
+            return
+
+        self._subplugins = []
+        for entry in self.SUBPLUGINS:
+            try:
+                subplugin = getattr(self.target, entry)
+                self._subplugins.append(subplugin)
+            except Exception:  # noqa
+                target.log.exception("Failed to load subplugin: %s", entry)
+
+    def _func(self, func_name: str) -> None:
+        """Return the supported subplugin records.
+
+        Args:
+            func_name: Exported function of the subplugin to find.
+
+        Yields:
+            Record from the sub function.
+        """
+        for subplugin in self._subplugins:
+            try:
+                for entry in getattr(subplugin, func_name)():
+                    yield entry
+            except Exception:
+                continue
+
+    def check_compatible(self) -> None:
+        if not len(self._subplugins):
+            raise UnsupportedPluginError("No compatible subplugins found")
+
+    def __init_subclass_namespace__(cls, **kwargs):
+        # If this is a direct subclass of a Namespace Plugin,
+        # create a reference to the current class for indirect subclasses
+        # so that the can autogenerate summary methods there
+        cls.__nsplugin__ = cls
+
+    def __init_subclass_subplugin__(cls, **kwargs):
+        # Does the direct subclass already have a SUBPLUGINS attribute?
+        # if not, create this attribute on the direct subclass
+        if not getattr(cls.__nsplugin__, "SUBPLUGINS", None):
+            cls.__nsplugin__.SUBPLUGINS = []
+
+        # Register the current plugin class as a subplugin with
+        # the direct subclass of NamespacePlugin
+        cls.__nsplugin__.SUBPLUGINS.append(cls.__namespace__)
+
+        # Collect the public attrs of the subplugin
+        for subplugin_func_name in get_nonprivate_attribute_names(cls):
+            subplugin_func = inspect.getattr_static(cls, subplugin_func_name)
+
+            # If we already have a summary method, skip
+            if getattr(cls.__nsplugin__, subplugin_func_name, None):
+                continue
+
+            # The attr need to be callable and exported
+            if not isinstance(subplugin_func, Callable) or not getattr(subplugin_func, "__exported__", False):
+                continue
+
+            # The method needs to have a single record descriptor as output
+            if not isinstance(getattr(subplugin_func, "__record__", None), RecordDescriptor):
+                continue
+
+            # The generic template for the aggregator method
+            def generate_aggregator(method_name):
+                def aggregator(self):
+                    yield from self._func(method_name)
+
+                aggregator.__name__ = "_aggr"
```

Comment:
What is this necessary for?

Reply:
Also removed, artifact of a previous solution.